### PR TITLE
Rename data-tid's of RadioFilterComponent

### PR DESCRIPTION
**Pull request by paudyalhari**
[Card](https://rentpath.leankit.com/card/596920021)

* Moved data-tid's from radioGroup input to label
* Updated data-tid's by appending with filter type

### CODE REVIEW

**Comment by tadjohnston** [comment]
I'd rather not add a `filterType`. This will set a precedent for adding `Desktkop` or `Mobile` for every component.  I think we should go over these changes.  We seem to be shoving things in rather than fixing the root of the issue.

File: packages/react-ui-core/src/Form/RadioGroup.js [hunk 1]
```diff
@@ -75,14 +75,16 @@ export default class RadioGroup extends Component {
...
   renderRadioButton(index, fieldProps) {
     const props = pick(this.props, ['name', 'hideInputElement', 'orientation'])
+    const label = { 'data-tid': 'radiogroup-radiobutton', text: fieldProps.label }
     return (
       <RadioButton
-        data-tid="radiogroup-radiobutton"
         key={`${this.id}-${index}`}
         checked={this.state.value === fieldProps.value}
         onChange={this.handleValueChange}
+        label={label}
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Form/RadioGroup.js]: You cannot change things like this in `react-ui-core`. This affects _everything_ that's using this component and you're now changing the structure of this label completely.  How does one add anything to label now? Won't this get overrided if I pass in a `label` prop?

File: packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js [hunk 2]
```diff
@@ -43,7 +45,18 @@ export default class RadioGroupFilterCard extends Component {
...
   get fields() {
-    return this.props.fields || []
+    const { fields = [], filterType = '' } = this.props
+    return fields.map(field => {
+      const obj = field
+
+      if (typeof field.label === 'string') {
+        obj.label = { text: field.label, 'data-tid': `radiogroup-radiobutton-${filterType}` }
+      } else {
+        obj.label = Object.assign({ 'data-tid': `radiogroup-radiobutton-${filterType}` }, field.label)
```
Comment (tadjohnston) [inline packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js]: What if the field is a function or a node? This doesn't work.

File: packages/react-ui-core/src/Form/RadioGroup.js (hunk 1 above)
Comment (tadjohnston) [reply packages/react-ui-core/src/Form/RadioGroup.js]: Also, this is now *forcing* a label which isn't always needed.

File: packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js [hunk 3]
```diff
@@ -43,7 +44,17 @@ export default class RadioGroupFilterCard extends Component {
...
   get fields() {
-    return this.props.fields || []
+    const { fields = [] } = this.props
+    return fields.map(field => {
+      const obj = field
+
+      if (typeof field.label === 'string') {
+        obj.label = { text: field.label, 'data-tid': `radiogroup-radiobutton-${field.label}` }
+      } else {
+        obj.label = Object.assign({ 'data-tid': `radiogroup-radiobutton-${field.label.text}` }, field.label)
```
Comment (tadjohnston) [inline packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js]: You still cannot do this.  What if `label` is empty in the field or it's not an object?

**Comment by tadjohnston** [comment]
Just put a `data-tid="label" on https://github.com/rentpath/react-ui/blob/master/packages/react-ui-core/src/Form/Label.js#L38 and call it good. 
```
    return (
      /* eslint-disable jsx-a11y/label-has-for */
      <label 
        data-tid="label"
        {...props} 
        className={classnames}
      >
        {text}{children}
      </label>
    )
```
The above allows label to be overwritten if one is passed in.  Also, by doing this it will only show up if label is used. In this PR, you're still _forcing_ label to be used even if it doesn't exist and that is incorrect.

File: packages/react-ui-core/src/Form/RadioGroup.js [hunk 4]
```diff
@@ -75,14 +75,15 @@ export default class RadioGroup extends Component {
 
   renderRadioButton(index, fieldProps) {
     const props = pick(this.props, ['name', 'hideInputElement', 'orientation'])
+
     return (
       <RadioButton
-        data-tid="radiogroup-radiobutton"
```
Comment (tadjohnston) [inline packages/react-ui-core/src/Form/RadioGroup.js]: I'm not sure why you removed this.

File: packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js [hunk 5]
```diff
@@ -78,9 +89,8 @@ export default class RadioGroupFilterCard extends Component {
       >
         <RadioGroup
           name="radio-group-filter-card-radio-group"
-          data-tid="radio-group-filter-card-radio-group"
```
Comment (tadjohnston) [inline packages/react-ui-ag/src/Filters/RadioGroupFilterCard.js]: 👍

File: packages/react-ui-core/src/Form/RadioGroup.js (hunk 4 above)
Comment (paudyalhari) [reply packages/react-ui-core/src/Form/RadioGroup.js]: I think I accidentally removed it ?

Comment (paudyalhari) [reply packages/react-ui-core/src/Form/RadioGroup.js]: Ohh ok I removed this to add data-tid dynamically, but seems like I need to revert this back.

Comment (tadjohnston) [reply packages/react-ui-core/src/Form/RadioGroup.js]: You know, I think maybe removing this is fine as long as there is one on the RadioButton already.

Comment (paudyalhari) [reply packages/react-ui-core/src/Form/RadioGroup.js]: We dont have that in RadioButton Field, I think it would be better to have it here then.

**Pull request closed without merging**